src/app/zap-templates/templates/app/cluster-objects-src.zapt

```diff
@@ -38,15 +40,54 @@ CHIP_ERROR DecodableType::Decode(TLV::TLVReader &reader) {
                 break;
         }
     }
-    VerifyOrReturnError(err == CHIP_END_OF_TLV, err);
+
     ReturnErrorOnFailure(reader.ExitContainer(outer));
     return CHIP_NO_ERROR;
 }
 
 } // namespace {{asType label}}
 {{/zcl_structs}}
-} // namespace {{asUpperCamelCase name}}
 
+namespace Commands {
+{{#zcl_commands}}
+namespace {{asUpperCamelCase name}} {
+CHIP_ERROR Type::Encode(TLV::TLVWriter &writer, uint64_t tag) const{
+    TLV::TLVType outer;
+    ReturnErrorOnFailure(writer.StartContainer(tag, TLV::kTLVType_Structure, outer));
+    {{#zcl_command_arguments}}
+    ReturnErrorOnFailure(DataModel::Encode(writer, TLV::ContextTag(k{{asUpperCamelCase label}}FieldId), {{asLowerCamelCase label}}));
+    {{/zcl_command_arguments}}
+    ReturnErrorOnFailure(writer.EndContainer(outer));
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR DecodableType::Decode(TLV::TLVReader &reader) {
+    CHIP_ERROR err;
+    TLV::TLVType outer;
+    VerifyOrReturnError(chip::TLV::kTLVType_Structure == reader.GetType(), CHIP_ERROR_WRONG_TLV_TYPE);
+    err = reader.EnterContainer(outer);
+    ReturnErrorOnFailure(err);
+    while ((err = reader.Next()) == CHIP_NO_ERROR) {
```

Comment:
So comparing this to our existing decoding code I see the following differences:

1. Our existing code ensures that all mandatory command arguments are present, or at least tries to.  I think it might be assuming all arguments are mandatory.  This is important, because our members don't have any initializers, generally, so if there are missing fields we will have un-initialized data in the struct.  Or we need to add initialization to some sort of (XML-defined?) default values....  Or something.
2. Our existing code errors out if the same field id is present twice.

Reply:
1. I think we need default value initializers. For starters, it can be type-specific defaults. I think this is more scalable path than trying to weed through mandatory vs. optional.
2. As I mentioned yesterday, I think we should not have the receiver do this checking, since it adds cost (which gets multiplied a fair bit). If its present twice, this logic will just continue without erroring and use the last specified value.

Reply:
https://github.com/project-chip/connectedhomeip/issues/10271